llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -1716,8 +1717,13 @@ bool TargetLowering::SimplifyDemandedBits(
       // if we don't care about FP signed-zero. The use of SETLT with FP means
       // that we don't care about NaNs.
       if (CC == ISD::SETLT && Op1.getValueType().isInteger() &&
-          (isNullConstant(Op1) || ISD::isBuildVectorAllZeros(Op1.getNode())))
+          (isNullConstant(Op1) || ISD::isBuildVectorAllZeros(Op1.getNode()))) {
+        if (Flags.hasSameSign()) {
+          Flags.setSameSign(false);
+          Op->setFlags(Flags);
```

Comment:
This doesn't seem necessary for the plumbing and its not clear its correct to propagate the rest of the flags.

Reply:
Don't we want to stop propagating the flag in `SimplifyDemandedBits`?

Reply:
Ahh, I misunderstood earlier.

I don't think we need to drop `Op` flags here. We aren't simplifying `Op0` or `Op1` here, just replacing `Op` with `Op0`.

Reply:
That makes sense, dropped it.